static/app/utils/replays/frame.tsx

```diff
@@ -0,0 +1,253 @@
+import {ReactNode} from 'react';
+
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconWarning} from 'sentry/icons';
+import {t, tct} from 'sentry/locale';
+import {BreadcrumbType} from 'sentry/types/breadcrumbs';
+import {TabKey} from 'sentry/utils/replays/hooks/useActiveReplayTab';
+import {
+  BreadcrumbFrame,
+  ErrorFrame,
+  LargestContentfulPaintFrame,
+  MutationFrame,
+  NavigationFrame,
+  SlowClickFrame,
+  SpanFrame,
+} from 'sentry/utils/replays/types';
+import type {Color} from 'sentry/utils/theme';
+
+export function getColor(frame: BreadcrumbFrame | SpanFrame | ErrorFrame): Color {
+  if ('category' in frame) {
+    switch (frame.category) {
+      case 'issue':
+        return 'red300';
+      case 'ui.slowClickDetected':
+        return (frame as SlowClickFrame).data.endReason === 'timeout'
+          ? 'red300'
+          : 'yellow300';
+      case 'replay.mutations':
+        return 'yellow300';
+      case 'ui.click':
+      case 'ui.input':
+      case 'ui.keyDown':
+      case 'ui.blur':
+      case 'ui.focus':
+        return 'purple300';
+      case 'console':
+      default: // Custom breadcrumbs will fall through here
+        return 'gray300';
+    }
+  }
+
+  switch (frame.op) {
+    case 'navigation.navigate':
+    case 'navigation.reload':
+    case 'navigation.back_forward':
+    case 'navigation.push':
+      return 'green300';
+    case 'largest-contentful-paint':
+    case 'memory':
+    case 'paint':
+    case 'resource.fetch':
+    case 'resource.xhr':
+    default:
+      return 'gray300';
+  }
+}
+
+export function getBreadcrumbType(
+  frame: BreadcrumbFrame | SpanFrame | ErrorFrame
+): BreadcrumbType {
```

Comment:
Can we add a note here that this is currently only used to determine what icon to show (`<BreadcrumbIcon>`)?